common/djangoapps/student/roles.py

```diff
@@ -49,7 +49,7 @@ def has_role(self, role, course_id, org):
         """
         return any(
             access_role.role == role and
-            access_role.course_id == course_id and
+            unicode(access_role.course_id) == unicode(course_id) and
```

Comment:
The issue was this 
```course.id``` is unicode object but ``ccx.id`` is not uni code it is number [views.py](https://github.com/edx/edx-platform/blob/master/lms/djangoapps/ccx/views.py#L135) So when we create ccx_locator i was not fully unicode because ccx_id was number. That why [roles.py](https://github.com/edx/edx-platform/blob/master/common/djangoapps/student/roles.py#L52) was not able to compare give ccx_locator with ccx_locator save in ``CourseAccessRole`` . See content of input ccx_locator and database value below
(
'course_id: ', CCXLocator(u'FAST', u'CC101', u'dummy_dec_2015', None, None, 10L), 
''access_role.course_id: ', CCXLocator(u'FAST', u'CC101', u'dummy_dec_2015', None, None, u'10'),
'is Equal: ', False
)

Fix this issue by making ccx.id unicode